### DISCUSSION TITLE
Mimir Compactor is deployed with Stateful Set, but Doc says its should be stateless

### DISCUSSION THREAD

**Discussion by mInrOz** [grafana/mimir#3060] [Help and support] [answered]
https://github.com/grafana/mimir/blob/3dee1a4629502b0324538b0b8424e736063e723a/operations/mimir/compactor.libsonnet#L5

https://grafana.com/docs/mimir/latest/operators-guide/architecture/components/compactor/
"The compactor is stateless."

**Comment by pracucci** [comment]
The compactor is stateless but it does quite a lot of disk I/O. We decided to deploy it as a StatefulSet to mount a dedicated volume for each replica, so that the compactor disk I/O doesn't impact the root filesystem.

**Reply by mInrOz** [reply]
Thanks for the clarification. Maybe create a quick note regarding this in the documentation or Jsonnet repo?
Its not a big deal, but some might react and or question why the compacter which is stateless is deployed as a stateful application.

**Comment by aknuds1** [comment, marked as answer]
@mInrOz I added an [issue](https://github.com/grafana/mimir/issues/3061) to clarify in docs. Thanks for bringing this to our attention :)

**Reply by mInrOz** [reply]
Thanks! And thanks again for the clarification @pracucci :)
